addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -55,6 +87,159 @@ def _compute_priority(self):
             if new_priority != batch.priority:
                 batch.priority = new_priority
 
+    @api.multi
+    @api.constrains('user_id')
+    def _compute_state(self):
+        """ Compute the state of a batch post confirm
+            waiting     : At least some picks are not ready
+            ready       : All picks are in ready state (assigned)
+            in_progress : All picks are ready and a user has been assigned
+            done        : All picks are complete (in state done or cancel)
+
+            the other two states are draft and cancel are manual
+            to transitions from or to the respective state.
+        """
+        for batch in self:
+
+            if batch.state in ['draft', 'done', 'cancel']:
+                # Can not do anything with them don't bother trying
+                continue
+
+            ready_picks, other_picks = \
+                self._calculate_pick_groups(batch.picking_ids)
+
+            if not batch.picking_ids or not (ready_picks or other_picks):
+                # This accounts for all pickings being complete, canceled or
+                # all pickings get removed for some reason.
+                # The last two can happen in any state
+                batch.state = 'done'
+
+            # Attempt transitions based on state
```

Comment:
I see the point of this comment, but it just ruins the `if` / `elif` indentation. Please remove the empty lines; you can't tell whether they go together at a glance.